tools/V4-CSDL-to-OpenAPI.xsl

```diff
@@ -4244,7 +4248,15 @@
               </xsl:if>
             </xsl:when>
             <xsl:otherwise>
-              <xsl:text>value</xsl:text>
+              <xsl:choose>
+                <xsl:when test="$odata-version='4.0'">
+                  <xsl:text>@odata.count":{"type":"number","description":"The number of entities in the collection. Available when using the [$count](http://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part1-protocol.html#sec_SystemQueryOptioncount) query option."},"value</xsl:text>
+                </xsl:when>
+                <xsl:otherwise>
+                  <!-- $odata-version > '4.0' -->
+                  <xsl:text>@count":{"type":"number","description":"The number of entities in the collection. Available when using the [$count](http://docs.oasis-open.org/odata/odata/v4.01/odata-v4.01-part1-protocol.html#sec_SystemQueryOptioncount) query option."},"value</xsl:text>
+                </xsl:otherwise>
+              </xsl:choose>

```

Comment:
Please create a reuse structure next to `components/schemas/error` and use
```
<xsl:text>@</xsl:text>
<xsl:if test="$odata-version='4.0'">
  <xsl:text>odata.</xsl:text>
</xsl:if>
<xsl:text>count":{$ref:'#/components/schemas/count'},"value</xsl:text>
``` 
here.

Reply:
> And the count can be a number or a string.

The [docs](http://docs.oasis-open.org/odata/odata-json-format/v4.01/odata-json-format-v4.01.html#sec_ControlInformationcountodatacount) states that it's a number

> The count control information occurs only in responses and can annotate any collection, see [OData-Protocol] section 11.2.5.5 System Query Option $count. **Its value is an Edm.Int64 value** corresponding to the total count of members in the collection represented by the request.

Reply:
> 3.2 [Controlling the Representation of Numbers](https://docs.oasis-open.org/odata/odata-json-format/v4.01/cos01/odata-json-format-v4.01-cos01.html#sec_ControllingtheRepresentationofNumber)
The IEEE754Compatible=true format parameter indicates that the service MUST serialize Edm.Int64 and Edm.Decimal numbers (including the count, if requested) as strings. This is in conformance with [RFC7493].

Reply:
oneOf or anyOf? I've seen this

```json
"Maximum": {
    "anyOf": [
        {
            "type": "number"
        },
        {
            "type": "string"
        }
    ],
```

but after reading the swagger documentation
> oneOf matches exactly one subschema, and anyOf can match one or more subschemas.

I think oneOf is correct as it can't be both a string and a number

Reply:
I guess it dones't really matter. I'll use anyOf for consistency